app/scripts/modules/kubernetes/src/container/environmentFrom.component.html

```diff
@@ -43,7 +43,8 @@
           <div class="col-md-3">
             <input type="checkbox"
                    style="margin-top: 10px; margin-left: 8px;"
-                   ng-model="envFrom.configMapRef.optional"/>
+                   ng-model="envFrom.configMapRef.optional"
+                   ng-init="envFrom.configMapRef.optional = ((envFrom.configMapRef.optional === null || envFrom.configMapRef.optional === undefined) || envFrom.configMapRef.optional)"/>
```

Comment:
Might be less surprising for developers if this logic was contained in an angular controller.

Reply:
Also it's acceptable in JS to use `myVar == null` (rather than triple equals) to check if a variable is either `null` or `undefined`. Though not sure if angular templates fully respect JS syntax.